### ISSUE TITLE
Document the need for `-t` when using socat as an ambassador

### ISSUE THREAD

**Issue by aidanhs** [docker/docker#16602]
https://groups.google.com/d/msg/docker-user/E_9o6L4FgOI/RW13qGuzAwAJ

This either needs a doc fix (to https://docs.docker.com/articles/ambassador_pattern_linking/) or a code tweak (to https://github.com/docker/docker/blob/2ed0739c7a47f5accfbae0c8cce2a7e6463dbdc6/api/client/hijack.go#L231).

While the Docker code itself is not buggy, I wonder if fixing 6 of the top 10 ambassador containers on the hub is a good enough reason? If you let me know the preferred route, I can raise a PR.

**Comment by cpuguy83** [comment]
Hold on, I do not see how ambassador containers has anything to do with this.
The OP was having issues using docker exec over socat, not some ambassador.

**Comment by aidanhs** [comment]
I'm the OP :)

The reality is that, as I noted in my e-mail, 6 of the top 10 results on the docker hub for 'ambassador' use socat and do not use `-t`. This means that all of these are broken for client connections that close the write side of the tcp connection and wait for a response for more than half a second.

I feel this is an issue with the docker documentation because the only documented ambassador container also uses socat without `-t`! https://docs.docker.com/articles/ambassador_pattern_linking/#the-svendowideit-ambassador-dockerfile

I've opened a PR at #16801 which also updates the ambassador container.

To briefly come back my original suggestion of possibly changing docker exec - I've changed my mind and think it's fine as-is - it's not technically doing anything wrong.
